covariancematrix.go

```diff
@@ -42,16 +42,18 @@ func CovarianceMatrix(cov *mat64.Dense, x mat64.Matrix, wts []float64) *mat64.De
 		floats.AddConst(-mean, v)
 	}
 
-	var xc mat64.Dense
-	xc.TCopy(&xt)
-
 	var n float64
 	if wts != nil {
 		if wr := len(wts); wr != r {
 			panic(mat64.ErrShape)
 		}
 
+		var xc mat64.Dense
+		xc.TCopy(&xt)
+
 		// Weight the rows.
+		// TODO(jonlawlor): if all weights are positive, we can still
+		// use mulgen if we take the square root.
```

Comment:
Could we do this too? The formula doesn't make sense with negative weights. I'm not sure where we want to document it, but no negative weights should be a general rule of this package.